javascript-practice/src/js/controllers/product-detail.controller.js

```diff
@@ -16,15 +17,50 @@ export default class ProductDetailController {
   async displayProductDetail() {
     const { params } = findRoute(window.location.pathname);
 
-    const product = await this.service.getById(params.id);
+    const product = await this.productService.getById(params.id);
 
     this.view.renderProduct(product);
+    
+    this.bindAddCartBtnEvent();
   }
 
   /**
    * Binds events to handle add product to cart
    */
   bindAddCartBtnEvent() {
+    const addCartItemBtnElement = getElementById('btn-add-cart');
+
+    addCartItemBtnElement.addEventListener('click', async () => {
+      const id = getElementById('product-id').value;
+
+      const product = await this.productService.getById(id);
+      const { name, colors, price, imgUrl, amount: productAmount } = product;
+
+      const existingCartItem = await this.cartService.getByProductId(id);
+      const { cartItemAmount } = existingCartItem;
+
+
+      if(productAmount <= 0) {
+        return showToastify('The product is running out of stock', 'toastify-danger');
+      }
+
+      // Needs to implement edit cart item first to expand this
+      // Update item amount if user already add it to cart
+      if(existingCartItem) {
```

Comment:
Should check condition before destructing obj `const { cartItemAmount } = existingCartItem;`